diff --git a/image_similarity/main.py b/image_similarity/main.py
index 9ca6660..9a33d98 100644
--- a/image_similarity/main.py
+++ b/image_similarity/main.py
@@ -58,6 +58,6 @@
 
 if __name__ == "__main__":
     logger.info("starting server")
-    server = WSGIServer(("0.0.0.0", 8002), app)
+    server = WSGIServer(("0.0.0.0", int(os.environ['IMAGE_SIMILARITY_SERVER_PORT'])), app)
     server_thread = gevent.spawn(server.serve_forever)
     gevent.joinall([server_thread])
